src/Sharpliner/AzureDevOps/Model/Step.cs

```diff
@@ -54,13 +54,22 @@ public abstract record Step
     /// </summary>
     [YamlIgnore]
     [DisallowNull]
-    public Conditioned<TimeSpan>? Timeout { get; init; }
+    // public Conditioned<TimeSpan>? Timeout { get; init; }
+    public Conditioned<TimeSpan>? Timeout
```

Comment:
I wonder if it's doable to just keep the Timeout visible to years? 
Maybe make the other internal?

So that when we set the Conditioned here, we would recreate the same one but return the int inside? Or that's too complicated?

Reply:
I'm not sure it's possible to have complex condition here

![image](https://github.com/user-attachments/assets/f8133f52-8672-48a1-8fed-92e0453c57f1)
```cs
new InlineBashTask("./.dotnet/dotnet test src/MySolution.sln")
{
    DisplayName = "Run unit tests",
    ContinueOnError = true,
    TimeoutInMinutes = If.Equal(variables["main_run"], "true")
        .Value(30)
        .Else
        .Value(parameters["timeout_in_minutes"])
}
```
this doesn't compile

Reply:
Ok, so maybe not to have a full support for an if but what I meant is to not surface two different fields, I think that might be confusing. But rather redirect the `Conditioned<TimeSpan>` so that it serializes correctly (maybe we can somehow re-create the `Conditioned<int>` out of it?

Reply:
@Meir017 this works for instance:
```csharp
    public Conditioned<int>? TimeoutInMinutes => Timeout switch {
        ConditionedParameterReference<TimeSpan> parameter => new ConditionedParameterReference<int>(parameter.Parameter),
        ConditionedVariableReference<TimeSpan> variable => new ConditionedVariableReference<int>(variable.Variable),
        _ => Timeout?.Definition != null ? (int)Timeout.Definition.TotalMinutes : null
    };
```

But it's a quick solution, incomplete, sort of.